apps/src/code-studio/levels/match.js

```diff
@@ -10,86 +15,174 @@ jQuery.fn.swap = function(b) {
+            // remove offset coordinates from this item
+            ui.draggable.css({top: 'auto', left: 'auto'});
+          } else {
+            // when an answer is in the rightmost list of answers, it can be dragged in to replace an empty slot
+            // in the central list of slots.
 
-          // this class is no longer in the answer list
-          movingItem.removeClass('answerlist');
+            var movingItem = ui.draggable.detach();
 
-          // this class can now be both dragged and a drop target for fellow answers in slots
-          movingItem.addClass('answerslot');
+            // replace target with this new item
+            $(event.target).replaceWith(movingItem);
 
-          // this new item can now be dropped onto by other answers in the central list
-          movingItem.droppable({
-            accept: '.answerslot',
-            activeClass: 'active',
-            drop: function(event, ui) {
-              if (enableSounds) {
-                CDOSounds.play('whoosh');
-              }
+            // the new item is now droppable
+            movingItem.droppable();
 
-              // remove offset coordinates from the dragged item
-              ui.draggable.css({top: '0px', left: '0px'});
+            // remove offset coordinates from the dragged item
+            movingItem.css({top: 'auto', left: 'auto'});
 
-              // determine y difference between old location and new location of item that will be swapped out
-              var origY = $(event.target).offset().top;
-              var destY = $(ui.draggable).offset().top;
-              var diffY = destY - origY;
+            // this class is no longer in the answer list
+            movingItem.removeClass('answerlist');
 
-              // swap this answer with the answer dropped onto it
-              ui.draggable.swap(event.target);
+            // this class can now be both dragged and a drop target for fellow answers in slots
+            movingItem.addClass('answerslot');
 
-              // move the target object back to its old location for a moment
-              $(event.target).css({top: -diffY + 'px'});
+            // this new item can now be dropped onto by other answers in the central list
+            this.makeItemDroppable(movingItem);
 
-              // and animate back to its new location
-              $(event.target).animate({top: '0px'});
+            // Once all answers have been dropped into a slot, let anyone
+            // listening know that an answer has been selected.
+            if ($(container).find('.match_answers .answer').length === 0) {
+              onAnswerChanged(this.levelId, true);
             }
-          });
+          }
         }
+      });
+  }
+
+  makeItemDroppable(item) {
+    const enableSounds = this.standalone;
+    const container = document.getElementById(this.id);
+    item.droppable({
+      accept: element =>
+        $(element).is('.answerslot') && $(container).find(element[0]).length,
```

Comment:
`makeItemDroppable` enables answers in the middle column as drop targets so that answers in the middle column can be swapped with each other. This change to the `accept` param makes it so that only the middle column *within the same level* will be highlighted (similar to the existing `accept` param above). It looks like this:
![drag-swap-2](https://user-images.githubusercontent.com/8001765/56158162-2ac76100-5f76-11e9-96a0-a544350c7bd3.gif)

The rest of the draggable/droppable config is not modified in this PR, just rearranged.